src/main/java/http/router/Router.java

```diff
@@ -26,6 +29,16 @@ public void addRoute(String path, List<httpMethod> methods, IController handler)
         routes.put(path, route);
     }
 
+    public void addRouteWithAuth(String path, List<httpMethod> methods, IController handler, IAuth auth) {
+        Route route = new Route(path, methods, handler, auth);
+        routes.put(path, route);
```

Comment:
Why does this method need me to pass in a path, if the route object already contains the path var??